### DISCUSSION TITLE
Dynamic relation load

### DISCUSSION THREAD

**Discussion by simplo-marcind** [prisma/prisma#24143] [Q&A] [answered]
### Question

Hi,
currently in prisma we have to pass relations which we will use during fetch object from database
`const user = await prisma.user.findFirst({
  include: {
    posts: true,
  },
})`

we can use the Proxy class in javascript 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy

Currently I use Eloquent ORM in PHP and it's i VERY helpful
https://laravel.com/docs/master/eloquent-relationships#one-to-many-inverse

Can we implement it in prisma? 

Have a nice day
Marcin

### Expected behavior (optional)

New solution would be like below:
`const user = await prisma.user.findFirst()
console.log(user.posts)`

**Comment by ludralph** [comment, marked as answer]
Hi @simplo-marcind 👋 

Prisma indeed requires you to specify the relations you want to include when fetching data from the database. As for your question about implementing a feature similar to Eloquent ORM's automatic resolution of inverse relationships, as of now, Prisma does not support this feature. In Prisma, you need to explicitly define the relations in your schema and also specify them in your queries.

However, if you want to see this implemented, you need to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) detailing your usecase so that our engineers can take a look.

**Comment by simplo-marcind** [comment]
HI @ludralph 
Thanks for your reply, I've created the issue https://github.com/prisma/prisma/issues/24149
